### ISSUE TITLE
WSL install fails with errors on Windows 10 guest VM

### ISSUE THREAD

**Issue by sirredbeard** [Microsoft/WSL#2872]
***** Your Windows build number:**  10.0.16299.15**

*** What you're doing and what's happening:** 

I installed Ubuntu from the Windows Store.

![screen shot 2018-01-23 at 3 17 55 pm](https://user-images.githubusercontent.com/33820650/35298596-a8726324-0050-11e8-8696-bd2c7994eb6f.png)

When I click Launch, I get this. It's 0x8007007e. I tried searching that error code, nothing very helpful.

![screen shot 2018-01-23 at 3 18 03 pm](https://user-images.githubusercontent.com/33820650/35298636-cf94ad04-0050-11e8-8b3e-902070314046.png)

I tried enabling WSL in Windows Features:

![screen shot 2018-01-23 at 3 09 33 pm](https://user-images.githubusercontent.com/33820650/35298688-f4257842-0050-11e8-89b8-2b1421960b9f.png)
![screen shot 2018-01-23 at 3 09 40 pm](https://user-images.githubusercontent.com/33820650/35298692-f78d53f6-0050-11e8-8050-6a7d89e3b501.png)

Which appears to work and then request restart:

![screen shot 2018-01-23 at 3 09 47 pm](https://user-images.githubusercontent.com/33820650/35298705-0473f16a-0051-11e8-8f1a-c46613003bd2.png)
![screen shot 2018-01-23 at 3 09 57 pm](https://user-images.githubusercontent.com/33820650/35298711-065995c0-0051-11e8-8cc7-f4dd87f22c1c.png)

But when I restart, it's disabled again:

![screen shot 2018-01-23 at 3 10 48 pm](https://user-images.githubusercontent.com/33820650/35298727-12abd266-0051-11e8-86c4-9fe77e5b7a04.png)
 
Attempting to enable the feature via PowerShell gives me this error message:

![screen shot 2018-01-23 at 3 12 19 pm](https://user-images.githubusercontent.com/33820650/35298749-23335370-0051-11e8-89c8-af7c4ab1fbbc.png)

I am running Windows 10 Educational edition in VMware Fusion 10.1.1 (7520154) on macOS 10.13.2 (17C205) on a MacBook Pro 2015.

The VMware image is configured to support hypervisors in the Guest.

![screen shot 2018-01-23 at 3 26 18 pm](https://user-images.githubusercontent.com/33820650/35298967-d4538cba-0051-11e8-8976-de8459508f5b.png)

*** What's wrong / what should be happening instead:**

Users cannot install WSL on VMware Fusion guests.

**Comment by sirredbeard** [comment]
**Attempted Troubleshooting**

- Uninstalling and reinstalling Ubuntu app
- Reinstalling WSL from Windows Features
- Restarting
- Ensuring virtualization enabled in VMware Fusion
- Enabling Developer Mode
- Installing OpenSUSE Leap 42
![screen shot 2018-01-23 at 3 46 59 pm](https://user-images.githubusercontent.com/33820650/35299860-aabbd7e2-0054-11e8-8582-031db7369b2d.png)
![screen shot 2018-01-23 at 3 47 15 pm](https://user-images.githubusercontent.com/33820650/35299871-b5844f1a-0054-11e8-9cd6-d2382382f915.png)
- Running bash and lxrun in cmd in Admin Mode
![screen shot 2018-01-23 at 3 45 00 pm](https://user-images.githubusercontent.com/33820650/35299808-6a5dc746-0054-11e8-96df-d3963086e16c.png)

**Comment by sirredbeard** [comment]
Here is the VMware hardware information. 

[msinfo.nfo.txt](https://github.com/Microsoft/WSL/files/1657576/msinfo.nfo.txt)

**Comment by bitcrazed** [comment]
Thanks for the great repro capture & sorry you're seeing this.

If LxRun.exe is not available, it looks like WSL itself isn't getting enabled on your machine! We'll get back to you soon.

**Comment by therealkenc** [comment]
Ref #2832 #2851 #2818 et al, albeit with slightly different modes of failure. It isn't going to be a "Guest VM" thing (that's my setup here; it works fine). We're flying multiple layers above with this fail. As a quick data point, is the Windows OS in your VM activated? `slmgr /xpr` or:
```
PS> Get-CimInstance -ClassName SoftwareLicensingProduct | where PartialProductKey | select Name, ApplicationId, LicenseStatus
```
Not saying that's the problem; just eliminating a variable. For some reason it is raining "one or more parent features are disabled so parent feature can not be enabled", whereas we weren't seeing them in volume previously.

**Comment by sirredbeard** [comment]
@therealkenc 

Windows is fully activated with a license tied to my Microsoft account.

![screen shot 2018-01-23 at 5 05 18 pm](https://user-images.githubusercontent.com/33820650/35303443-f53bf788-005f-11e8-8bb3-869eb57a0b53.png)
![screen shot 2018-01-23 at 5 05 40 pm](https://user-images.githubusercontent.com/33820650/35303444-f54bfde0-005f-11e8-9312-2568d04b9711.png)
![screen shot 2018-01-23 at 5 06 28 pm](https://user-images.githubusercontent.com/33820650/35303445-f55a4a6c-005f-11e8-864b-882251cb5540.png)

It sounds most similar to reports in #2832, in particular [this one](https://github.com/Microsoft/WSL/issues/2832#issuecomment-357320923).

I am not getting the same error as in #2851, I get:

`'bash' is not recognized as an internal or external command,
operable program or batch file.`

and #2818 involves upgrading from legacy WSL to the store, I am trying to install WSL on a fresh Windows 10 install.

**Comment by bitcrazed** [comment]
@sirredbeard - could you please email me: richturn at you know where dot com?

**Comment by sirredbeard** [comment]
I performed a clean install of Windows 10, enabled WSL from Windows Features before installing Ubuntu from the Store, and have been unable to duplicate the problem.

If the issue occurs again, I will file requested logs.

**Comment by bitcrazed** [comment]
Thanks for letting us know.

If you **or anyone else** can repro this issue, please capture the logs below and reach out to `taraj at you know where dot com` to arrange receipt of said logs so that we can diagnose.

TIA.

**Comment by sirredbeard** [comment]
@bitcrazed 

Unfortunately I had wiped the old image before I got your request for logs. I need a bigger hard drive.

**Comment by bitcrazed** [comment]
LOL. No problem. Thanks for reporting anyhow. 

We do have an internally tracked issue that _may_ describe the root cause of the issue. When fixed, this should resolve at least one possible cause ;)

**Comment by ideal2545** [comment]
I'm also having this issue, any luck with a fix? I just can't launch bash.exe it just sits at a black screen, and then eventually I get an error. I look and I can't seem to get the service to run.

**Comment by Biswa96** [comment]
Which virtualization program do you use? I can install WSL in VirutalBox VM (haven't tried Store Apps).

**Comment by bitcrazed** [comment]
@ideal2545 - 

1. Do you have WSL enabled & have you rebooted since enabling? Run the following from an elevated Cmd/PowerShell prompt to double-check you've got WSL enabled:
``` powershell
Get-WindowsOptionalFeature -FeatureName microsoft-windows-subsystem-linux -online
```
2. Do you have 3rd party Anti Virus installed?
3. What do you see when you run `wslconfig /l` from Cmd/PowerShell?

**Comment by ideal2545** [comment]
@Biswa96 I am using VMware fusion pro (8.5 and 10)

@bitcrazed 
1. WSL is enabled based on that. 
2. No Anti Virus
3. wslconfig /l - results in error: 0x800880005 

P.S. I have been able to reproduce this on two machines now - both from clean/scratch without too much configuration.

**Comment by papersaltserver** [comment]
I have had the same problem on a physical machine having Insider Build 10.0.18356.1
All the symptoms were equal: feature enabled, lxrun not found, wslconfig /l shows no distros. https://aka.ms/wslstore link not working.
To fix this, I've found Ubuntu in Windows Store and installed it. Right after installation, nothing worked as well. Then I've opened Windows Store once again and pressed Launch for Ubuntu there. It has done some initialization and "bash" command started to work after that, however "lxrun" still shows "not found"
